skimage/exposure/exposure.py

```diff
@@ -18,7 +18,42 @@
                     'float': dtype_range[np.float64]})
 
 
-def histogram(image, nbins=256):
+def _offset_array(arr, low_boundary, high_boundary):
+    """Offset the array to get the lowest value at 0 if negative."""
+    if low_boundary < 0:
+        offset = low_boundary
+        dyn_range = high_boundary - low_boundary
+        # get smallest dtype that can hold both minimum and offset maximum
+        offset_dtype = np.promote_types(np.min_scalar_type(dyn_range),
+                                        np.min_scalar_type(low_boundary))
+        if arr.dtype != offset_dtype:
+            # prevent overflow errors when offsetting
+            arr = arr.astype(offset_dtype)
+        arr = arr - offset
+    else:
+        offset = 0
+    return arr, offset
+
+
+def _bincount_histogram(image, source_range):
+    """Efficient histogram calculation for a flat image of integers."""
```

Comment:
Can you document the inputs and outputs to this function? Even though it is not public, as a developer it is nicer not to have to read the source code (or get a ValueError) to know that `source_range` can only be "image" or "dtype".